src/actions/organizationActions.js

```diff
@@ -59,39 +59,46 @@ const determineSelectedOrganization = (organizations, selectedOrganization) => {
   return firstOrganization;
 };
 
-// organizationsLoad
-// -----------------
-// This long function does various requests to the Giant Swarm API
-// and massages the responses into some reasonable state for Happa to
-// work with.
-//
+/**
+ * This action does various requests to the Giant Swarm API
+ * and massages the responses into some reasonable state for Happa to
+ * work with.
+ */
 export function organizationsLoad() {
   return function(dispatch, getState) {
-    var organizationsApi = new GiantSwarm.OrganizationsApi();
+    const organizationsApi = new GiantSwarm.OrganizationsApi();
 
-    var alreadyFetching = getState().entities.organizations.isFetching;
+    const alreadyFetching = getState().entities.organizations.isFetching;
 
     if (alreadyFetching) {
-      return new Promise(resolve => {
-        resolve();
-      });
+      return new Promise(resolve => resolve());
     }
 
     dispatch({ type: types.ORGANIZATIONS_LOAD });
 
     return organizationsApi
       .getOrganizations()
       .then(organizations => {
-        var organizationsArray = organizations.map(organization => {
-          return organization.id;
-        });
+        const organizationsArray = organizations.map(
+          organization => organization.id
+        );
 
-        var orgDetails = Promise.all(
+        const orgDetails = Promise.all(
           organizationsArray.map(organizationName => {
             return organizationsApi
               .getOrganization(organizationName)
               .then(organization => {
-                return organization;
+                return organizationsApi
+                  .getCredentials(organization.id)
+                  .then(credentials => {
+                    const organizationWithCredentials = Object.assign(
+                      {},
+                      organization,
+                      { credentials }
+                    );
+
+                    return organizationWithCredentials;
```

Comment:
Maybe, probably, you have a reason for this, IDK, but there are two things that catch my attention here:

1.  Nesting. If we really want to have a clear data flow as discussed, we should avoid nesting as much as possible. Nesting makes it difficult to read. BTW No blaming here: I am guilty of this as well, especially in actions files. Just let's stop doing this.

Best approach, as I we have it described here: https://github.com/giantswarm/giantswarm/issues/7521 IMO is to have a separate method that does this, for reusability and readability. 

2. I think we already have a method for this, maybe we should refactor it? Or maybe this is another thing we will hav to have in mind when doing the data flow refactor

Reply:
I just added to what was already there, since I know there's a refactor coming.

To avoid nesting, which is indeed difficult to read, we can use async await, which I'll also do here to illustrate my point.

The function we currently have loads credentials for a given id and then sets the credentials for the current organization in the state

Reply:
> just added to what was already there, since I know there's a refactor coming.

If that's not a lot of hassle, I think we shouldn't continue writing stuff the old way, cause in the end it will cause more work during the refactor